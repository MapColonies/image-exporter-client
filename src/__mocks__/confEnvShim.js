if (!window._env_) {
  window._env_ = (function (undefined) {
    return {
      LANGUAGE: 'en',
      MAP_SERVER: 'MAP_SERVER',
      PUBLISH_POINT: 'PUBLISH_POINT',
      CHANNEL: 1002,
      VERSION: 1,
      REQUEST: 'REQUEST',
      SERVICE_PROTOCOL: 'SERVICE_PROTOCOL',
      SERVICE_NAME: 'SERVICE_NAME',
      ACTIVE_LAYER: 'ACTIVE_LAYER',
      BBOX: {
        limit: 20000,
      },
      EXPORT: {
        defaultZoomLevel: 18,
        maxZoomLevel: 22
      },
      MAP: {
        center: [35.14, 31.39],
        zoom: 8,
      },
      ACTIVE_LAYER_PROPERTIES: {
        urlPattern : 'MAP_SERVER/service',
        exportUrlPattern: 'MAP_SERVER/service?REQUEST=REQUEST&SERVICE=SERVICE&LAYERS=LAYER',
        urlPatternParams: {
          service: 'SERVICE',
          layers: 'LAYER',
          tiled: 'TILED',
          matrixSet: 'MATRIXSET',
          style: 'STYLE',
          projection: 'PROJECTION'
        }
      },
    };
  })(void 0);
}
