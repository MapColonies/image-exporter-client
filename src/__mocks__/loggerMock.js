export default {
  log: () => {},
  info: () => {},
  error: () => {},
  debug: () => {},
};
