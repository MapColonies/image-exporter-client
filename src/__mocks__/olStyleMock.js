export class Fill {
  constructor() { }
}

export class Stroke {
  constructor() { }
}

export class Circle {
  constructor() { }
}

export class Style {
  constructor() { }
}
