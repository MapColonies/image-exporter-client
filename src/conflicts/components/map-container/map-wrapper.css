.ol-zoom {
  top: 2.5em;
  right: 0.5em;
  left: unset;
}

body[dir="rtl"] .ol-full-screen {
  right: unset;
  left: .5em;
}

body[dir="rtl"] .ol-zoom {
  right: unset;
  left: 0.5em;
}

.mdc-dialog__title + .mdc-dialog__content{
  padding-top: 20px;
}

.mdc-dialog__actions{
  padding-right: 24px;
}
