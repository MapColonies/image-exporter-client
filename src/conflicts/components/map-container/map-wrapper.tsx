import React from 'react';
import { Geometry } from 'geojson';
import rewind from '@turf/rewind';
import { Polygon } from 'geojson';
import { Map,
  TileLayer,
  VectorSource,
  GeoJSONFeature,
  TileOsm,
  VectorLayer,
  DrawInteraction,
  DrawType
 } from '@map-colonies/react-components';
import './map-wrapper.css';

interface MapWrapperProps {
  drawType?: DrawType;
  selectionPolygon?: Polygon;
  onPolygonSelection: (polygon: Polygon) => void;
}

export const MapWrapper: React.FC<MapWrapperProps> = (props) => {
  const handlePolygonSelected = (geometry: Geometry): void => {
    const rewindedPolygon = rewind(geometry as Polygon);
    props.onPolygonSelection(rewindedPolygon);
  };

  return (
    <Map allowFullScreen={true} showMousePosition={true}>
      <TileLayer>
        <TileOsm />
      </TileLayer>
      {props.selectionPolygon && (
        <VectorLayer>
          <VectorSource>
            <GeoJSONFeature geometry={props.selectionPolygon} />
          </VectorSource>
        </VectorLayer>
      )}
      {props.children}
      {props.drawType !== undefined && (
        <DrawInteraction
          drawType={props.drawType}
          onPolygonSelected={handlePolygonSelected}
        />
      )}
    </Map>
  );
};
