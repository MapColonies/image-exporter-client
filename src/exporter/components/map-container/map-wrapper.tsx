import React from 'react';
import { Geometry } from 'geojson';
import rewind from '@turf/rewind';
import { Polygon } from 'geojson';
import { Map,
  VectorSource,
  GeoJSONFeature,
  VectorLayer,
  DrawInteraction,
  DrawType
 } from '@map-colonies/react-components';
import EXPORTER_CONFIG from '../../../common/config';
import './map-wrapper.css';

interface MapWrapperProps {
  drawType?: DrawType;
  selectionPolygon?: Polygon;
  onPolygonSelection: (polygon: Polygon) => void;
}

export const MapWrapper: React.FC<MapWrapperProps> = (props) => {
  const handlePolygonSelected = (geometry: Geometry): void => {
    const rewindedPolygon = rewind(geometry as Polygon);
    props.onPolygonSelection(rewindedPolygon);
  };

  return (
    <Map 
      allowFullScreen={true}
      showMousePosition={true}
      zoom={EXPORTER_CONFIG.MAP.ZOOM}
      center={EXPORTER_CONFIG.MAP.CENTER}
    >
      {props.selectionPolygon && (
        <VectorLayer>
          <VectorSource>
            <GeoJSONFeature geometry={props.selectionPolygon} />
          </VectorSource>
        </VectorLayer>
      )}
      {props.children}
      {props.drawType !== undefined && (
        <DrawInteraction
          drawType={props.drawType}
          onPolygonSelected={handlePolygonSelected}
        />
      )}
    </Map>
  );
};
