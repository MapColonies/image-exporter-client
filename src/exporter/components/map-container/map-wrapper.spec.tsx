import React from 'react';
import { shallow } from 'enzyme';
import { Polygon } from 'geojson';
import { DrawInteraction, DrawType, VectorLayer } from '@map-colonies/react-components';
// eslint-disable-next-line
import '../../../__mocks__/confEnvShim';
import { MapWrapper } from './map-wrapper';


const polygonSelection = jest.fn();
const selectionPolygon:Polygon = {
  type: 'Polygon',
  coordinates: [
    [ [100.0, 0.0], [101.0, 0.0], [101.0, 1.0],
      [100.0, 1.0], [100.0, 0.0] ]
    ]
  };

describe('MapWrapper component', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <MapWrapper
        children={(<span>test</span>)}
        onPolygonSelection={polygonSelection}
        drawType={DrawType.BOX}
        selectionPolygon={selectionPolygon}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('VectorLayer not rendered when no selectionPolygon', () => {
    const wrapper = shallow(
      <MapWrapper
        children={(<span>test</span>)}
        onPolygonSelection={polygonSelection}
        drawType={DrawType.BOX}
        selectionPolygon={undefined}
      />
    );

    expect(wrapper.find(VectorLayer)).toEqual({}); 
  });

  it('DrawInteraction not rendered when no DrawType defined', () => {
    const wrapper = shallow(
      <MapWrapper
        children={(<span>test</span>)}
        onPolygonSelection={polygonSelection}
        drawType={undefined}
        selectionPolygon={undefined}
      />
    );

    expect(wrapper.find(DrawInteraction)).toEqual({}); 
  });

  it('children SPAN rendered with proper content', () => {
    const childrenContent = 'test';
    const chilComponent = <span>{childrenContent}</span>;
    const wrapper = shallow(
      <MapWrapper
        children={chilComponent}
        onPolygonSelection={polygonSelection}
        drawType={undefined}
        selectionPolygon={undefined}
      />
    );

    expect(wrapper.find('span')).toBeDefined();
    expect(wrapper.find('span').prop('children')).toEqual(childrenContent); 
  });


});
