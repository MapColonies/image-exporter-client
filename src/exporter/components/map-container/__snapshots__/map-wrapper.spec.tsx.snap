// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapWrapper component renders correctly 1`] = `
<Component
  allowFullScreen={true}
  center={
    Array [
      35.14,
      31.39,
    ]
  }
  showMousePosition={true}
  zoom={8}
>
  <Component>
    <Component>
      <Component
        geometry={
          Object {
            "coordinates": Array [
              Array [
                Array [
                  100,
                  0,
                ],
                Array [
                  101,
                  0,
                ],
                Array [
                  101,
                  1,
                ],
                Array [
                  100,
                  1,
                ],
                Array [
                  100,
                  0,
                ],
              ],
            ],
            "type": "Polygon",
          }
        }
      />
    </Component>
  </Component>
  <span>
    test
  </span>
  <Component
    drawType="BOX"
    onPolygonSelected={[Function]}
  />
</Component>
`;
